awsiot/iotshadow.py

```diff
@@ -1699,3 +1779,451 @@ def __init__(self, *args, **kwargs):
+            ],
+            response_paths = [
+                awscrt.mqtt_request_response.ResponsePath(
+                    accepted_topic,
+                    "clientToken"
+                ),
+                awscrt.mqtt_request_response.ResponsePath(
+                    rejected_topic,
+                    "clientToken"
+                )
+            ],
+            publish_topic = publish_topic,
+            payload = json.dumps(request.to_payload()).encode(),
+            correlation_token = correlation_token,
+        )
+
+        internal_unmodeled_future = self._rr_client.make_request(request_options)
+
+        return awsiot.create_v2_service_modeled_future(internal_unmodeled_future, "delete_named_shadow", accepted_topic, DeleteShadowResponse, V2ErrorResponse)
+
+    def delete_shadow(self, request : DeleteShadowRequest) -> concurrent.futures.Future :
+        """
+
+        Deletes the (classic) shadow for an AWS IoT thing.
+
+        API Docs: https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html#delete-pub-sub-topic
+
+        Args:
+            request: `DeleteShadowRequest` instance.
+
+        Returns:
+            A Future whose result will be an instance of `DeleteShadowResponse`.
+        """
+        request._validate()
+
+        publish_topic = '$aws/things/{0.thing_name}/shadow/delete'.format(request)
+        accepted_topic = publish_topic + "/accepted";
+        rejected_topic = publish_topic + "/rejected";
+
+        subscription0 = '$aws/things/{0.thing_name}/shadow/delete/+'.format(request);
+
+        correlation_token = str(uuid.uuid4())
+        request.client_token = correlation_token
+
+        request_options = awscrt.mqtt_request_response.RequestOptions(
+            subscription_topic_filters = [
+                subscription0,
+            ],
+            response_paths = [
+                awscrt.mqtt_request_response.ResponsePath(
+                    accepted_topic,
+                    "clientToken"
+                ),
+                awscrt.mqtt_request_response.ResponsePath(
+                    rejected_topic,
+                    "clientToken"
+                )
+            ],
+            publish_topic = publish_topic,
+            payload = json.dumps(request.to_payload()).encode(),
+            correlation_token = correlation_token,
+        )
+
+        internal_unmodeled_future = self._rr_client.make_request(request_options)
+
+        return awsiot.create_v2_service_modeled_future(internal_unmodeled_future, "delete_shadow", accepted_topic, DeleteShadowResponse, V2ErrorResponse)
+
+    def get_named_shadow(self, request : GetNamedShadowRequest) -> concurrent.futures.Future :
+        """
+
+        Gets a named shadow for an AWS IoT thing.
+
+        API Docs: https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html#get-pub-sub-topic
+
+        Args:
+            request: `GetNamedShadowRequest` instance.
+
+        Returns:
```

Comment:
Had discussion about whether we should report that in addition to the expected return, a "rejected" request response operation will complete the future exceptionally with a `V2ServiceException`. This documentation hint will help users understand and deal with potentially rejected operations.